### DISCUSSION TITLE
app.js createStore for Pinia not being generated

### DISCUSSION THREAD

**Discussion by sonicviz** [quasarframework/quasar#16852] [General - Components / Directives / etc]
Hi, 
I have a quasar prototype that was using Vuex and I'm converting it to Pinia. 

The .quasar/app.js file has "import createStore from 'app/src/store/index'" that seems generated from the vuex src/stores directory. 

I added pinia via NPM and created a store in src/stores, which is populated ok with the example store, index.js and store-flag.d.ts files.

However, running quasar dev the .quasar/app.js still only has  "import createStore from 'app/src/store/index'" and does not auto generate the equivalent "import createStore from 'app/src/stores/index'" for the pinia stores, so Pinia never gets initialised. 

Where and how is this enabled so it's picked up by quasar and inserted into the .quasar/app.js file so you can then use pinia?
I followed the guide at https://quasar.dev/quasar-cli-vite/state-management-with-pinia/ and on the pinia site, but can't seem to trigger it.
It says you can run vuex and pinia side by side, so my expectation here is that the correct pinia import from src/stores will be generated in .quasar/app.js just like it is for vuex src/store, but it's not happening. 

Creating a fresh base project with just pinia from the cli it generates aok, which I used as a test comparison.

Any clues to get Pinia working properly?

Thanks!

**Comment by yusufkandemir** [comment]
You can use Vuex and Pinia side by side, but only one of them gets integrated natively. When you have both packages installed, Vuex is preferred. That decision has been made, mainly to keep the structure simple and not introduce any breaking changes. You need to use a boot file to initialize Pinia. But, in boot/router callback parameters, `store` would be the Vuex instance. As soon as you complete the migration and delete the Vuex package, you can start enjoying the usual integration with Pinia.

**Reply by sonicviz** [reply]
Thanks. I uninstalled Vuex and deleted the store folder but app.js was still not getting the pinia import inserted. 
I'll try it again deleting node_modules as noted below though. 

Do you have an example boot file for pinia?

**Reply by TobyMosque** [reply]
the boot itself looks the same:

**vuex**
```js
export default boot(async ({ store }) => {
  await store.dispatch('sample/setup', { /* payload */ })
})
```

**pinia**
```js
import { useSampleStore } from 'stores/sample'
export default boot(async ({ store }) => {
  const sampleStore = useSampleStore(store)
  await sampleStore.setup({ /* payload */ })
})
```

**Reply by sonicviz** [reply]
aok, thanks!

**Comment by TobyMosque** [comment]
Quasar CLI will only manage one store, vuex or pinia, since vuex come first, if both are in the package vuex will be priorized.

If u wanna to keep both, u'll need to handle pinia by yourself, if u wanna to let Quasar CLI manage pinia, u would uninstall vuex (and probably delete node_modules and reinstall).

The reason behind that, is quasar can't inject both stores in the boots, preFetch, etc.

**Reply by sonicviz** [reply]
Thanks. I uninstalled Vuex and deleted the store folder but app.js was still not getting the pinia import inserted.

I didn't reinstall node_modules though, so could be something to do with that I guess.

**Reply by TobyMosque** [reply]
Quasar detects vuex by search for a node_modules/vuex folder,

u can try:
```
1 - delete `node_modules/vuex`
2 - delete `node_modules`, so rerun `npm i` or `yarn`
3 - delete `node_modules` and lock files, so rerun `npm i` or `yarn`
```

**Reply by sonicviz** [reply]
aha. Ok, I'll delete/reinstall node_modules and see if that fixes it.

Thanks.

**Reply by sonicviz** [reply]
Ok, that works, thanks. 

I have a subsequent issue though. 

I'm doing an authentication flow using a BaaS Appwrite, and need to do an authentication call in the Router/index.js to check if the user still has a connected session. If I do this as a helper import, it works aok ie: call to a computed property isLogged that's checked in the routes for authenticated vs unauthenticated redirection. 

However, if I try and do this via the pinia store as a property I get the following error thrown just on the imports in router/index.js
import { useAuthStore } from "../stores/auth";
const auth = useAuthStore()

pinia.mjs:1709 Uncaught (in promise) Error: [🍍]: "getActivePinia()" was called but there was no active Pinia. Are you trying to use a store before calling "app.use(pinia)"?

This is the same error I was getting before when Pinia wasn't being created in the .quasar/app.js , which is now working. 

I guess this is an execution order type issue in setup?
How can access the pinia store in the router/index.js?

Thanks!

**Reply by TobyMosque** [reply]
https://pinia.vuejs.org/core-concepts/outside-component-usage.html

u would pass the pinia instance to the store, as i did in the boot.
router/index also has a wrap who receives the store as a parameter, very similar to the boot.
![image](https://github.com/quasarframework/quasar/assets/13981214/9d59b9ec-c864-48ba-adab-bf020c498376)

**Reply by sonicviz** [reply]
Aok, I'll give that a shot. Thanks for the help, both of you!
